flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -165,7 +165,7 @@ protected ClusterEntrypoint(Configuration configuration) {
 
         shutDownHook =
                 ShutdownHookUtil.addShutdownHook(
-                        this::cleanupDirectories, getClass().getSimpleName(), LOG);
+                        () -> this.closeAsync().join(), getClass().getSimpleName(), LOG);
```

Comment:
I think it would be great to also test this bit here. One idea could be to use the `TestProcessBuilder` to create a separate process. This process should then be killed via `SIGTERM`. We would then need something which waits for a little bit before doing the cleanup. The cleanup is ideally something like deleting a file which can be verified from the other process. What do you think?

Reply:
IIUC, you are suggesting to add a unit test to guard the `closeAsync()` is executed in the `shutdownHook`. Right?

Reply:
Yes exactly.